packages/inventory/src/Stocktake/api/hooks.ts

```diff
@@ -76,19 +76,30 @@ export const useInsertStocktake = () => {
   });
 };
 
+export const useUpdateStocktake = () => {
+  const queryClient = useQueryClient();
+  const api = useStocktakeApi();
+  return useMutation(api.update, {
+    onSuccess: () => {
+      return queryClient.invalidateQueries(['stocktake']);
+    },
+  });
+};
+
 export const useStocktakeFields = <
   KeyOfStocktake extends keyof StocktakeFragment
 >(
   keys: KeyOfStocktake | KeyOfStocktake[]
 ): FieldSelectorControl<StocktakeFragment, KeyOfStocktake> => {
   const { stocktakeNumber = '' } = useParams();
+  const { mutateAsync } = useUpdateStocktake();
   const { data } = useStocktake();
   const api = useStocktakeApi();
   return useFieldsSelector(
     ['stocktake', stocktakeNumber],
     () => api.get.byNumber(Number(stocktakeNumber)),
     (patch: Partial<StocktakeFragment>) =>
-      api.update({ ...patch, id: data?.id ?? '' }),
+      mutateAsync({ ...patch, id: data?.id ?? '' }),
```

Comment:
realised this fn (in each of the different features) wasn't invalidating.. so if you were really a big meany you could update a stocktake comment for example, navigate back to the list view and it wouldn't refetch as the list query wasn't invalidated.